packages/trpc/server/routers/viewer/workflows/util.ts

```diff
@@ -114,3 +250,587 @@ export async function removeSmsReminderFieldForBooking({
+          emailBody: step.reminderBody || "",
+          template: step.template,
+          sender: step.sender,
+          workflowStepId: step.id,
+          prisma,
+        });
+      } else if (step.action === WorkflowActions.SMS_NUMBER && step.sendTo) {
+        await scheduleSMSReminder({
+          evt: bookingInfo,
+          reminderPhone: step.sendTo,
+          triggerEvent: trigger,
+          action: step.action,
+          timeSpan: {
+            time,
+            timeUnit,
+          },
+          message: step.reminderBody || "",
+          workflowStepId: step.id,
+          template: step.template,
+          sender: step.sender,
+          userId: userId,
+          teamId: teamId,
+          prisma: prisma,
+        });
+      } else if (step.action === WorkflowActions.WHATSAPP_NUMBER && step.sendTo) {
+        await scheduleWhatsappReminder({
+          evt: bookingInfo,
+          reminderPhone: step.sendTo,
+          triggerEvent: trigger,
+          action: step.action,
+          timeSpan: {
+            time,
+            timeUnit,
+          },
+          message: step.reminderBody || "",
+          workflowStepId: step.id || 0,
+          template: step.template,
+          userId: userId,
+          teamId: teamId,
+        });
+      }
+    });
+    await Promise.all(promiseScheduleReminders);
+  });
+  return Promise.all(promiseSteps);
+}
+
+export function isStepEdited(oldStep: WorkflowStep, newStep: WorkflowStep) {
+  const oldStepKeys = Object.keys(oldStep);
+  const newStepKeys = Object.keys(newStep);
+
+  if (oldStepKeys.length !== newStepKeys.length) {
+    return true;
+  }
+
+  for (const key of oldStepKeys) {
+    if (oldStep[key as keyof WorkflowStep] !== newStep[key as keyof WorkflowStep]) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+export async function getAllWorkflowsFromEventType(
+  eventType: {
+    workflows?: {
+      workflow: WorkflowType;
+    }[];
+    teamId?: number | null;
+    parentId?: number | null;
+    parent?: {
+      teamId: number | null;
+    } | null;
+    metadata?: Prisma.JsonValue;
+  } | null,
+  userId?: number | null
+) {
+  if (!eventType) return [];
+
+  const eventTypeWorkflows = eventType?.workflows?.map((workflowRel) => workflowRel.workflow) ?? [];
+
+  const teamId = await getTeamIdFromEventType({
+    eventType: {
+      team: { id: eventType?.teamId ?? null },
+      parentId: eventType?.parentId ?? null,
+    },
+  });
+
+  const orgId = await getOrgIdFromMemberOrTeamId({ memberId: userId, teamId });
+
+  const isManagedEventType = !!eventType?.parentId;
+
+  const eventTypeMetadata = EventTypeMetaDataSchema.parse(eventType?.metadata || {});
+
+  const workflowsLockedForUser = isManagedEventType
+    ? !eventTypeMetadata?.managedEventConfig?.unlockedFields?.workflows
+    : false;
+
+  const allWorkflows = await getAllWorkflows(
+    eventTypeWorkflows,
+    userId,
+    workflowsLockedForUser ? eventType?.parent?.teamId : teamId,
```

Comment:
Can we simply not pass `teamId` which is aware of getting the teamId from parent for a managed event?